CONTRIBUTING.md

```diff
@@ -81,39 +81,58 @@ cherry-picked into `book-v2` branch.
+Similarly, the introduction of a new major version (`v(X+1).Y`) of the Go plugin might only lead to a new minor (`vX.(Y+1).Z`) release of KubeBuilder, since no breaking change is being made to the CLI itself. It'd only be a breaking change to KubeBuilder if we remove support for an older version of the plugin.
 
-**NOTE:** In the case of the `v1` and `v2` PROJECT version, a corresponding Plugin version is implied and constant -- `v1` implies the `go.kubebuilder.io/v1` Plugin, and similarly for `v2`.  This is for legacy purposes -- no such implication is made with the `v3` PROJECT version.
+For more information on how the release and plugin versions work, see the [semver](https://semver.org/) documentation.
 
-## Introducing changes in the scaffold files
+**NOTE:** schema for project versions `"1"` and `"2"` were defined before the concept of plugins was introduced, so plugin `go.kubebuilder.io/v2.0` is implicitly used for those project types. Schema for project versions `"3-alpha"` and beyond define a `layout` key that informs the plugin system of which plugin to use.
 
-Changes in the scaffolded files require a new Plugin version. If we delete or update a file that is scaffolded by default, it's a breaking change and requires a `MAJOR` Plugin version.  If we add a new file, it may not be a breaking change.
+## Introducing changes to plugins
 
-**More simply:** any change that will break the expected behaviour of a project built with the previous `MINOR` Plugin versions is a breaking change to that plugin. 
+Changes made to plugins may require a plugin version increase depending on whether
+they break the current version or not. See the plugins design doc [versioning section][plugins-design-versioning]
+for more details.
 
-**EXAMPLE:**
+Breaking changes require either a new Kubebuilder _minor_ or _major_ release.
+A new _major_ release is only necessary for changes that fundamentally alter
+all `kubebuilder` projects.
 
-KubeBuilder Release version (`5.3.1`) scaffolds projects with the plugin version `3.2.1` by default.
+Non-breaking changes do not require a release.
 
-The changes introduced in our PR will not work well with the projects which were built with the plugin versions `3.0.0...3.5.1` without users taking manual steps to update their projects. Thus, our changes introduce a breaking change to the Go plugin, and require a `MAJOR` Plugin version bump.
+### Example
 
-In the PR, we should add a migration guide to the [Migrations](https://book.kubebuilder.io/migrations.html) section of the KubeBuilder book. It should detail the required steps that users should take to upgrade their projects from `go.kubebuilder.io/3.X.X` to the new `MAJOR` Plugin version `go.kubebuilder.io/4.0.0`.
+KubeBuilder release version `v2.3.1` scaffolds projects with plugin `go.kubebuilder.io/v2.1`
+by default. Previously, plugin `go.kubebuilder.io/v2.0` was used.
```

Comment:
I am afraid that use the `v2.3.1` which is a valid release brings confusing since because of this it the example was using a version that was not released yet to illustrate how it works.

Reply:
Personally I don't think the versions we use in an example matter.

Reply:
The text `KubeBuilder release version `v2.3.1` scaffolds projects with plugin `go.kubebuilder.io/v2.1` shows a fact because the release and version exists and lead the impression that the v2.X.X would just release plugins v2.X when both are not true. So, I'd suggest:

Reply:
Removed kubebuilder version from this section.